halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/ci/CiController.java

```diff
@@ -31,129 +30,118 @@
 import com.netflix.spinnaker.halyard.util.v1.GenericGetRequest;
 import com.netflix.spinnaker.halyard.util.v1.GenericUpdateRequest;
 import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
-@RestController
-@RequiredArgsConstructor
-@RequestMapping("/v1/config/deployments/{deploymentName:.+}/ci")
-public class CiController {
+public abstract class CiController<T extends Master, U extends Ci<T>> {
```

Comment:
Little confused about what's going on here with the API root (e.g. `/v1/config/deployments/...`). Can you clarify?

Reply:
Each sub-controller is declaring a specific path for that CI system.